### PR TITLE
[fix](nereids)need substitute agg function using agg node's output if it's in order by key

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by hello-stephen** [comment]
run performance

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FillUpMissingSlots.java [hunk 1]
```diff
@@ -183,10 +183,14 @@ static class Resolver {
         private final List<Expression> groupByExpressions;
         private final Map<Expression, Slot> substitution = Maps.newHashMap();
         private final List<NamedExpression> newOutputSlots = Lists.newArrayList();
+        private final Map<Slot, Expression> outputSubstitutionMap;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FillUpMissingSlots.java:186]:
```suggestion
        private final Map<Slot, Expression> outputSubstitutionMap = Maps.newHashMap();
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FillUpMissingSlots.java [hunk 2]
```diff
@@ -183,10 +183,14 @@ static class Resolver {
...
 
         Resolver(Aggregate aggregate) {
             outputExpressions = aggregate.getOutputExpressions();
             groupByExpressions = aggregate.getGroupByExpressions();
+            outputSubstitutionMap = outputExpressions.stream().filter(Alias.class::isInstance)
+                    .collect(Collectors.toMap(alias -> alias.toSlot(), alias -> alias.child(0),
+                            (k1, k2) -> k1));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/FillUpMissingSlots.java:193]: use foreach(outputSubstitutionMap.put(alias -> alias.toSlot(), alias -> alias.child(0)))

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**